architecture/adr-list/adr-002-identity-transactions.md

```diff
@@ -0,0 +1,272 @@
+8. **`service`** (optional): A set of Service Endpoint maps
+9. **`@context`** (optional): A list of strings
+
+**Example:**
+
+```javascript
+{
+  "@context": [
+    "https://www.w3.org/ns/did/v1",
+    "https://w3id.org/security/suites/ed25519-2020/v1"
+  ],
+  "id": "N22KY2Dyvmuu2PyyqSFKue",
+  "authentication": [
+    {
+      "id": "N22KY2Dyvmuu2PyyqSFKue#z6MkecaLyHuYWkayBDLw5ihndj3T1m6zKTGqau3A51G7RBf3",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "N22KY2Dyvmuu2PyyqSFKue",
+      "publicKeyMultibase": "zAKJP3f7BD6W4iWEQ9jwndVTCBq8ua2Utt8EEjJ6Vxsf"
+    }
+  ],
+  "capabilityInvocation": [
+    {
+      "id": "N22KY2Dyvmuu2PyyqSFKue#z6MkhdmzFu659ZJ4XKj31vtEDmjvsi5yDZG5L7Caz63oP39k",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "N22KY2Dyvmuu2PyyqSFKue",
+      "publicKeyMultibase": "z4BWwfeqdp1obQptLLMvPNgBw48p7og1ie6Hf9p5nTpNN"
+    }
+  ],
+  "capabilityDelegation": [
+    {
+      "id": "N22KY2Dyvmuu2PyyqSFKue#z6Mkw94ByR26zMSkNdCUi6FNRsWnc2DFEeDXyBGJ5KTzSWyi",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "N22KY2Dyvmuu2PyyqSFKue",
+      "publicKeyMultibase": "zHgo9PAmfeoxHG8Mn2XHXamxnnSwPpkyBHAMNF3VyXJCL"
+    }
+  ],
+  "assertionMethod": [
+    {
+      "id": "N22KY2Dyvmuu2PyyqSFKue#z6MkiukuAuQAE8ozxvmahnQGzApvtW7KT5XXKfojjwbdEomY",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "N22KY2Dyvmuu2PyyqSFKue",
+      "publicKeyMultibase": "z5TVraf9itbKXrRvt2DSS95Gw4vqU3CHAdetoufdcKazA"
+    }
+  ]
+ }
+```
+
+#### Verification Method
+
+1. **`id`** (string): A string with format `<DIDDoc-id>#<key-alias`
+2. **`controller`**: A list of base58-encoded identifier strings.
+3. **`type`** (string)
+4. **`publicKeyJwk`** (`map[string,string]`, optional): A map representing a JSON Web Key that conforms to [RFC7517](https://tools.ietf.org/html/rfc7517). See definition of `publicKeyJwk` for additional constraints.
+5. **`publicKeyMultibase`** (optional): A base58-encoded string that conforms to a [MULTIBASE](https://datatracker.ietf.org/doc/html/draft-multiformats-multibase-03) encoded public key.
+
+**Example:**
+
+```javascript
+{
+  "id": "N22KY2Dyvmuu2PyyqSFKue#key-0",
+  "type": "JsonWebKey2020",
+  "controller": "N22KY2Dyvmuu2PyyqSFKue",
+  "publicKeyJwk": {
+    "kty": "OKP", // external (property name)
+    "crv": "Ed25519", // external (property name)
+    "x": "VCpo2LMLhn6iWku8MKvSLg2ZAoC-nlOyPVQaO3FxVeQ" // external (property name)
+}
+```
+
+#### Service
+
+1. **`id`** (string): The value of the id property MUST be a URI conforming to [RFC3986](https://www.rfc-editor.org/rfc/rfc3986). A conforming producer MUST NOT produce multiple service entries with the same ID. A conforming consumer MUST produce an error if it detects multiple service entries with the same ID. It has a follow format: `<DIDDoc-id>#<service-alias>`
```

Comment:
No, a service doesn't need to follow always being a DIDDoc at the start because it's possible to have non-DID services.

Reply:
A good example of this is available if you look up did:sov:WRfXPg8dantKVubE3HX8pw on https://dev.uniresolver.io/

You'll see here the services are defined as:
- agent: https://agents.danubeclouds.com/agent/WRfXPg8dantKVubE3HX8pw
- xdi: https://xdi03-at.danubeclouds.com/cl/+!:did:sov:WRfXPg8dantKVubE3HX8pw

Another example on ION is if you look up: did:ion:EiClkZMDxPKqC9c-umQfTkR8vvZ9JPhl_xLDI9Nfk38w5w where the services are:
- linkedin: linkedin.com/in/henry-tsai-6b884014
- github: github.com/thehenrytsai

Reply:
https://www.w3.org/TR/did-core/#services